### PR TITLE
fix(generate): error instead of clobbering when task-docs markers are missing

### PR DESCRIPTION
**Pull request by JamBalaya56562**
## Summary
- `mise generate task-docs --inject` now requires both `<!-- mise-tasks -->` and `<!-- /mise-tasks -->` and leaves the file untouched when they are missing, instead of destroying it
- the end marker is searched *after* the start marker, so a reversed pair is an error rather than a silent replace-to-EOF
- extracted the injection into a pure `inject_task_docs()` helper with unit tests, plus an e2e regression test

## Context
The panic reported in https://github.com/jdx/mise/discussions/4676 (both markers present) was fixed in #5651. While confirming that for the reporter, the same code path turned out to still mangle files that *don't* have the markers.

```rust
let start = contents.find(task_placeholder_start).unwrap_or(0);          // no marker -> 0
let end = contents[start..].find(task_placeholder_end)
    .map(|e| e + start).unwrap_or(contents.len());                       // no marker -> EOF
contents.replace_range((start + task_placeholder_start.len())..end, &doc); // always cuts at byte 19
```

With no start marker, `start` is `0` and the replacement begins at byte 19 of a file that has no marker at all.

Reproduced with the released 2026.7.14 binary:

| input | result before this PR |
|---|---|
| README with no markers | truncated to its first 19 bytes, then the docs appended — `# My Project\n\nImpor## \`hello\` …` |
| file shorter than the marker (`hi\n`) | panic: `range start index 19 out of range for slice of length 3` |
| start marker but no end marker | everything from the marker to EOF replaced, no warning |
| end marker before the start marker | same silent replace-to-EOF |
| multibyte file where byte 19 is mid-character | char-boundary panic |

Errors now name the file and the marker that's missing, e.g.:

```
mise ERROR README.md does not contain the `<!-- mise-tasks -->` marker required by --inject, add:

<!-- mise-tasks -->
<!-- /mise-tasks -->
```

The happy path is byte-for-byte the same computation as before, so `e2e/generate/test_generate_task_docs_inject` is unchanged and still passes as the regression check.

## Tests
- unit tests on `inject_task_docs()`: happy path, missing start marker, missing end marker, reversed markers, file shorter than the marker, multibyte content
- `e2e/generate/test_generate_task_docs_inject_errors`: each failure mode asserts a non-zero exit, an error mentioning the missing marker, no `panicked` in the output, and that the file's contents are byte-identical afterwards

## Notes
- the `--inject` long help gained one line ("The file must already contain both comments; mise errors instead of modifying the file if they are missing."); `mise.usage.kdl`, `docs/cli/generate/task-docs.md` and `man/man1/mise.1` were updated to match, so the `mise run render` diff check should stay clean
- I could not compile locally (no C toolchain on that box), so the before/after table above was captured with the released 2026.7.14 binary and the code change itself is validated by the tests in CI

Relates to https://github.com/jdx/mise/discussions/4676

*This PR was prepared by an AI coding assistant.*

## Summary by CodeRabbit

* **Bug Fixes**
  * `mise generate task-docs --inject` now fails with a clear error when the `<!-- mise-tasks -->` / `<!-- /mise-tasks -->` markers are missing, reversed, or otherwise malformed, instead of risking incorrect replacement.
  * Prevents unintended modifications in edge cases involving short files or multibyte content.

* **Documentation**
  * Updated CLI help and manuals to clarify that both marker comments must already exist for injection to work.

* **Tests**
  * Added e2e coverage to verify failure behavior and that target files remain byte-identical on errors.

### CODE REVIEW

File: e2e/generate/test_generate_task_docs_inject_errors [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+# markers in the wrong order
+reversed='<!-- /mise-tasks -->
+
+body
+
+<!-- mise-tasks -->'
+echo "$reversed" >reversed.md
+assert_fail_contains "mise generate task-docs --inject --output reversed.md" "<!-- /mise-tasks -->"
+assert "cat reversed.md" "$reversed"
```
[the remark this answers is not in the document]
Comment (JamBalaya56562) [reply e2e/generate/test_generate_task_docs_inject_errors]: Both points were valid — fixed in b941f6e.

All five failure cases now go through one helper:

```bash
assert_inject_leaves_file_alone() {
  local file="$1" expected="$2" out
  cp "$file" "$file.orig"
  out="$(quiet_assert_fail "mise generate task-docs --inject --output $file")"
  assert_contains_text "$out" "$expected"
  assert_not_contains_text "$out" "panicked"
  if cmp -s "$file" "$file.orig"; then
    ok "[$file] left unchanged"
  else
    fail "[$file] was modified by a failed --inject"
  fi
}
```

- the `|| true` is gone; `quiet_assert_fail` asserts the non-zero exit for every case, including the multibyte one, and the output is reused for both the marker and the no-panic assertion
- `cmp -s` against a pre-command copy replaces `assert "cat …"`, so trailing bytes are part of the comparison
- the fixtures are heredocs now rather than `echo "$var"`, for the same reason

*This reply was generated by an AI coding assistant.*

File: e2e/generate/test_generate_task_docs_inject_errors [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# shorter than the marker: this used to panic with
+# "range start index 19 out of range for slice of length 3"
+echo "hi" >short.md
+assert_fail_contains "mise generate task-docs --inject --output short.md" "<!-- mise-tasks -->"
+assert_not_contains "mise generate task-docs --inject --output short.md 2>&1 || true" "panicked"
+assert "cat short.md" "hi"
+
+# multibyte content, where byte 19 is not a char boundary
+echo "日本語のドキュメント" >multibyte.md
+assert_not_contains "mise generate task-docs --inject --output multibyte.md 2>&1 || true" "panicked"
```
[the remark this answers is not in the document]
Comment (JamBalaya56562) [reply e2e/generate/test_generate_task_docs_inject_errors]: Valid — fixed in b941f6e. The multibyte case no longer discards the exit status: it now runs through the same helper as the other failure cases, which uses `quiet_assert_fail` (asserts the non-zero exit), then asserts the missing-marker diagnostic and the absence of `panicked` in that same output, and finally `cmp -s` against a pre-command copy to prove the file was left byte-identical.

*This reply was generated by an AI coding assistant.*
